nilearn/surface/surface.py

```diff
@@ -1753,3 +1779,167 @@ def from_volume(
+    -------
+    vmin : float
+
+    vmax : float
+    """
+    vmin = min(min(x.ravel()) for x in img.data.parts.values())
+    vmax = max(max(x.ravel()) for x in img.data.parts.values())
+    return vmin, vmax
+
+
+def concat_extract_surface_data_parts(img):
+    """Concatenate the data of a SurfaceImage across hemispheres and return
+    as a numpy array.
+
+    Parameters
+    ----------
+    img : :obj:`~nilearn.surface.SurfaceImage` object
+        SurfaceImage whose data to concatenate and extract.
+
+    Returns
+    -------
+    :obj:`~numpy.ndarray`
+        Concatenated data across hemispheres.
+    """
+    return np.concatenate(list(img.data.parts.values()), axis=0)
+
+
+def concat_imgs(imgs):
+    """Concatenate the data of a list or tuple of SurfaceImages.
+
+    Assumes all images have same meshes.
+
+    Parameters
+    ----------
+    imgs : :obj:`list` or :obj:`tuple` of SurfaceImage object
+
+    Returns
+    -------
+    SurfaceImage object
+    """
+    if not isinstance(imgs, (tuple, list)) or any(
+        not isinstance(x, SurfaceImage) for x in imgs
+    ):
+        raise TypeError(
+            "'imgs' must be a list or a tuple of SurfaceImage instances."
+        )
+
+    if len(imgs) == 1:
+        return imgs[0]
+
+    for i, img in enumerate(imgs):
+        check_same_n_vertices(img.mesh, imgs[0].mesh)
+        imgs[i] = at_least_2d(img)
+
+    output_data = {}
+    for part in imgs[0].data.parts:
+        tmp = [img.data.parts[part] for img in imgs]
+        output_data[part] = np.concatenate(tmp, axis=1)
+
+    return new_img_like(imgs[0], data=output_data)
+
+
+def mean_img(img):
+    """Compute mean of SurfaceImage over time points (for 'time series').
+
+    Parameters
+    ----------
+    img : SurfaceImage
+
+    Returns
+    -------
+    SurfaceImage
+    """
+    if len(img.shape) < 2 or img.shape[1] < 2:
+        data = img.data
+    else:
+        data = {
+            part: np.mean(value, axis=1).astype(float)
+            for part, value in img.data.parts.items()
+        }
+    return new_img_like(img, data=data)
+
+
+def two_dim_to_one_dim(img):
+    """Deconcatenate a 2D Surface image into a a list of 1D SurfaceImages.
+
+    Parameters
+    ----------
+    img : SurfaceImage object
+
+    Returns
+    -------
+    :obj:`list` or :obj:`tuple` of SurfaceImage object
+    """
+    if not isinstance(img, SurfaceImage):
+        raise TypeError("Input must a be SurfaceImage.")
+
+    if len(img.shape) < 2 or img.shape[1] < 2:
+        return [img]
+
+    return [
+        new_img_like(img, data=_extract_surface_image_data(img, i))
+        for i in range(img.shape[1])
+    ]
+
+
+def _extract_surface_image_data(surface_image, index):
+    mesh = surface_image.mesh
+    data = surface_image.data
+
+    return {
+        hemi: data.parts[hemi][..., index]
+        .copy()
+        .reshape(mesh.parts[hemi].n_vertices, 1)
+        for hemi in data.parts
+    }
+
+
+def new_img_like(ref_niimg, data):
```

Comment:
maybe rename this variable?
```suggestion
def new_img_like(ref_img, data):
```